### PR TITLE
add searchPlace feature by merchant and city

### PR DESCRIPTION
**Pull request by willkoua**

### CODE REVIEW

File: src/main/java/io/tchepannou/enigma/profile/dao/StationRepository.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import io.tchepannou.enigma.profile.domain.Station;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+
+@Repository
+public interface StationRepository extends CrudRepository<Station, Integer> {
+
+    ArrayList<Station> findByMerchantAndCityId(Merchant merchant, Integer cityId);
```
Comment (htchepannou) [inline src/main/java/io/tchepannou/enigma/profile/dao/StationRepository.java]: replace ``ArrayList`` by ``List``
When declaring a collection, always use interfaces, rather than implementation

File: src/test/java/io/tchepannou/enigma/profile/controller/MerchantControllerIT.java [hunk 2]
```diff
@@ -122,4 +122,20 @@ public void findByIdShouldReturn404ForInvalidId() throws Exception {
...
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+
+                .andDo(MockMvcResultHandlers.print())
+                .andExpect(status().isOk())
+
+                .andExpect(jsonPath("$.placeIds", is(Arrays.asList(1,3))))
+        ;
+
+    }
```
Comment (htchepannou) [inline src/test/java/io/tchepannou/enigma/profile/controller/MerchantControllerIT.java]: add a test for invalid city

File: src/main/resources/db/migration/V1_7__add_table_station.sql [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+) ENGINE = InnoDB;
+
+INSERT INTO T_STATION(id, merchant_fk, place_id, city_id) VALUES
+  (1, 110, 1, 2370001),
+  (2, 120, 2, 2370002),
+  (3, 130, 3, 2370003),
+  (4, 140, 4, 2370004),
+  (5, 150, 5, 2370005),
+  (6, 160, 6, 2370006),
+  (7, 200, 7, 2370007)
```
Comment (htchepannou) [inline src/main/resources/db/migration/V1_7__add_table_station.sql]: - Make sure that each merchant has station in both yaoude/douala (arrival and departure)
- for improving readability, add data with merchant_fk=place_id. This way, its easier to correlate a merchant and its place.

The script should look like:
```
INSERT INTO T_STATION(merchant_fk, place_id, city_id) VALUES
(110, 1101, 2370001),
(110, 1102, 2370002),
(120, 1201, 2370001),
(120, 1202, 2370002),
...
```
- Make sure to add in another PR the details of those places in ref-data-service

File: src/main/resources/db/migration/V1_7__add_table_station.sql [hunk 4]
```diff
@@ -0,0 +1,19 @@
+CREATE TABLE T_STATION(
+    id              INT      NOT NULL AUTO_INCREMENT,
+    merchant_fk     INT      NOT NULL REFERENCES T_MERCHANT(id),
+
+    place_id        INT      NOT NULL,
+    city_id         INT      NOT NULL,
```
Comment (htchepannou) [inline src/main/resources/db/migration/V1_7__add_table_station.sql]: add a unique constraint between merchant/place and city

File: src/main/java/io/tchepannou/enigma/profile/service/StationService.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    @Autowired
+    private Mapper mapper;
+
+    public SearchPlaceResponse findPlacesByMerchantAndCity(Integer id, Integer cityId){
+        final Merchant merchant = merchantRepository.findOne(id);
+        if (merchant == null){
+            throw new NotFoundException(ProfileErrorCode.MERCHANT_NOT_FOUND);
+        }
+
+        Iterable<Station> listStations = stationRepository.findByMerchantAndCityId(merchant, cityId);
```
Comment (htchepannou) [inline src/main/java/io/tchepannou/enigma/profile/service/StationService.java]:
```
List<Station> listStations = ...
```

File: src/main/java/io/tchepannou/enigma/profile/service/StationService.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+        return new SearchPlaceResponse(
+                StreamSupport.stream(listStations.spliterator(), false)
```
Comment (htchepannou) [inline src/main/java/io/tchepannou/enigma/profile/service/StationService.java]: ... Then
```
listStation.stream()
....
```

File: src/main/resources/db/migration/V1_7__add_table_station.sql (hunk 3 above)
Comment (willkoua) [reply src/main/resources/db/migration/V1_7__add_table_station.sql]: je ne comprends pas très bien ce que tu veux que je fasse ici

File: src/main/java/io/tchepannou/enigma/profile/service/StationService.java (hunk 6 above)
Comment (willkoua) [reply src/main/java/io/tchepannou/enigma/profile/service/StationService.java]: je ne comprends pas

Comment (htchepannou) [reply src/main/java/io/tchepannou/enigma/profile/service/StationService.java]: you can use ``listStation.stream()`` (instead of StreamSupport.stream(...)`` if you change the type of ``listStation`` from ``Iterable``  to ``List``

File: src/main/resources/db/migration/V1_7__add_table_station.sql (hunk 3 above)
Comment (htchepannou) [reply src/main/resources/db/migration/V1_7__add_table_station.sql]: change the format for INSERT sql so that
- place_id=<merchant_id><city_suffix>.

Ex:
- merchant_id=110,  city_id=2370001 => city_suffix=1 => place_id=1101
- merchant_id=110,  city_id=2370002 => city_suffix=2 => place_id=1102
